fdbserver/workloads/MockDDTrackerShardEvaluator.actor.cpp

```diff
@@ -0,0 +1,186 @@
+/*
+ * MockDDTrackerShardEvaluator.actor.cpp
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2013-2022 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "fdbserver/workloads/MockDDTest.h"
+#include "flow/actorcompiler.h" // This must be the last #include.
+
+struct MockDDTrackerShardEvaluatorWorkload : public MockDDTestWorkload {
+	static constexpr auto NAME = "MockDDTrackerShardEvaluator";
+	DDSharedContext ddcx;
+
+	PromiseStream<RelocateShard> output;
+	PromiseStream<GetMetricsRequest> getShardMetrics;
+	PromiseStream<GetTopKMetricsRequest> getTopKMetrics;
+	PromiseStream<GetMetricsListRequest> getShardMetricsList;
+	PromiseStream<Promise<int64_t>> getAverageShardBytes;
+
+	KeyRangeMap<ShardTrackedData> shards;
+
+	ActorCollection actors;
+	uint64_t mockDbSize = 0;
+	const int keySize = 16;
+
+	std::map<int, int> rsReasonCounts;
```

Comment:
This can be a `std::map<RelocateReason, int>` if `RelocateReason::operator<` is defined.